bootstrap/kubeadm/api/v1alpha4/kubeadmconfig_webhook.go

```diff
@@ -114,8 +133,45 @@ func (c *KubeadmConfigSpec) validate(name string) error {
 		knownPaths[file.Path] = struct{}{}
 	}
 
-	if len(allErrs) == 0 {
-		return nil
+	return allErrs
+}
+
+func (c *KubeadmConfigSpec) validateIgnition() field.ErrorList {
```

Comment:
Following up on the discussion started [here](https://github.com/kubernetes-sigs/cluster-api/pull/4172#discussion_r575305296): I removed the check which forces specifying the `ignition` block when `format: ignition` is specified. The reason is - the `ignition` block allows specifying **extra** Ignition-related configuration, however if all the user wants to do is generate bootstrap data in the default Ignition format, it looks weird IMO to be forced to specify `ignition: {}`.

Reply:
This was an FYI comment from me. Since there doesn't seem to be any feedback, I guess this can be resolved.